harvester_e2e_tests/integration/test_hosts.py

```diff
@@ -392,72 +392,73 @@ def test_delete_vm_after_host_shutdown(
             f", script path: {host_state.path}"
         )
 
-    # Delete VM and check it been deleted
-    code, data = api_client.vms.delete(focal_vm['name'], focal_vm['namespace'])
-    assert 200 == code, (code, data)
-    spec = api_client.vms.Spec.from_dict(data)
-    endtime = datetime.now() + timedelta(seconds=wait_timeout)
-    while endtime > datetime.now():
-        code, data = api_client.vms.get(focal_vm['name'], focal_vm['namespace'])
-        if 404 == code:
-            break
-        sleep(3)
-    else:
-        raise AssertionError(
-            f"Failed to Delete VM({focal_vm['name']}) with errors:\n"
-            f"Status({code}): {data}"
-        )
-
-    fails, check = [], dict()
-    for vol in spec.volumes:
-        vol_name = vol['volume']['persistentVolumeClaim']['claimName']
-        check[vol_name] = api_client.volumes.delete(vol_name)
+    try:
+        # Delete VM and check it been deleted
+        code, data = api_client.vms.delete(focal_vm['name'], focal_vm['namespace'])
+        assert 200 == code, (code, data)
+        spec = api_client.vms.Spec.from_dict(data)
+        endtime = datetime.now() + timedelta(seconds=wait_timeout)
+        while endtime > datetime.now():
+            code, data = api_client.vms.get(focal_vm['name'], focal_vm['namespace'])
+            if 404 == code:
+                break
+            sleep(3)
+        else:
+            raise AssertionError(
+                f"Failed to Delete VM({focal_vm['name']}) with errors:\n"
+                f"Status({code}): {data}"
+            )
 
-    endtime = datetime.now() + timedelta(seconds=wait_timeout)
-    while endtime > datetime.now():
-        l_check = dict()
-        for vol_name, (code, data) in check.items():
-            if 200 != code:
+        fails, check = [], dict()
+        for vol in spec.volumes:
+            vol_name = vol['volume']['persistentVolumeClaim']['claimName']
+            check[vol_name] = api_client.volumes.delete(vol_name)
+
```

Comment:
Personally I think it's better to check delete 200 here for a more clear logic since we decouple different check (delete 200 and get 404) to respective section.

Reply:
It looks github showing wrong code snippet, yeah this should be enhanced.